### PR TITLE
[Merged by Bors] - feat(LocallyConvex/Bounded): add `isVonNBounded_iff_tendsto_smallSets_nhds`

### PR DESCRIPTION
**Pull request by urkud**
* add `absorbs_iff_eventually_nhds_zero`,
  `isVonNBounded_iff_tendsto_smallSets_nhds`,
  and `isVonNBounded_pi_iff`;
* generalize some lemmas from `NormedField` to `NormedDivisionRing`;
* use new lemmas to golf 2 proofs

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/LocallyConvex/Basic.lean [hunk 1]
```diff
@@ -183,6 +182,18 @@ theorem absorbent_iff_eventually_nhdsWithin_zero :
 
 alias ⟨Absorbent.eventually_nhdsWithin_zero, _⟩ := absorbent_iff_eventually_nhdsWithin_zero
 
+theorem absorbs_iff_eventually_nhds_zero (h₀ : 0 ∈ s) :
+    Absorbs 𝕜 s t ↔ ∀ᶠ c : 𝕜 in 𝓝 0, MapsTo (c • ·) t s :=
+  ⟨fun h ↦ h.eventually_nhds_zero h₀, fun h ↦ absorbs_iff_eventually_nhdsWithin_zero.2 <|
+    h.filter_mono inf_le_left⟩
```
Comment (ADedecker) [inline Mathlib/Analysis/LocallyConvex/Basic.lean:188]: Could you instead replace `Absorbs.eventually_nhds_zero` by
```lean
theorem absorbs_iff_eventually_nhds_zero (h₀ : 0 ∈ s) :
    Absorbs 𝕜 s t ↔ ∀ᶠ c : 𝕜 in 𝓝 0, MapsTo (c • ·) t s := by
  rw [absorbs_iff_eventually_nhdsWithin_zero, ← nhdsWithin_compl_singleton_sup_pure, 
      Filter.eventually_sup, Filter.eventually_pure, and_iff_left]
  intro x _
  simpa only [zero_smul]
```
and add an alias for the forward direction?

File: Mathlib/Analysis/LocallyConvex/Bounded.lean [hunk 2]
```diff
@@ -185,16 +191,8 @@ variable {𝕝 : Type*} [NormedField 𝕜] [NontriviallyNormedField 𝕝] [AddCo
 
 theorem IsVonNBounded.smul_tendsto_zero {S : Set E} {ε : ι → 𝕜} {x : ι → E} {l : Filter ι}
     (hS : IsVonNBounded 𝕜 S) (hxS : ∀ᶠ n in l, x n ∈ S) (hε : Tendsto ε l (𝓝 0)) :
```
Comment (ADedecker) [inline Mathlib/Analysis/LocallyConvex/Bounded.lean:193]: Looks like you can change `𝕜` to a `NormedDivisionRing` here as well.

**Review by ADedecker — APPROVED** [review]
Thanks a lot, all this golfing is very nice!

bors d+

File: Mathlib/Analysis/LocallyConvex/Bounded.lean (hunk 2 above)
Comment (urkud) [reply Mathlib/Analysis/LocallyConvex/Bounded.lean:193]: Indeed, more lemmas can be generalized. I have more in a local branch. I'm going to open a separate PR about that.

File: Mathlib/Analysis/LocallyConvex/Basic.lean (hunk 1 above)
Comment (urkud) [reply Mathlib/Analysis/LocallyConvex/Basic.lean]: I decided not to use `alias` to get `Absorbs` argument first. I think that it's more natural for dot notation lemmas.

**Pull request closed without merging**